studentSubmissions/script.js

```diff
@@ -0,0 +1,162 @@
+addSubmission(submissions, "Josh", 77);
+addSubmission(submissions, "Aaron", 59);
+addSubmission(submissions, "Troy", 95);
+console.log("New Submissions:", submissions);
+
+// specThree
+// Declare a function named deleteSubmissionByIndex
+// Parameter(s): array, index
+// Functionality: remove an object from the `array` at the specified `index` using the splice method.
+
+const deleteSubmissionByIndex = (array, index) => {
+    array.splice(index, 1);
+};
+deleteSubmissionByIndex(submissions, 4); // removes Josh at index 4 from the submissions
+deleteSubmissionByIndex(submissions, 4); // since Josh is removed, Aaron moves up in the index to the 4th spot
+console.log("Submissions After Deleting by Index", submissions);
+
+
+// specFour
+// Declare a function named deleteSubmissionByName
+// Parameter(s): array, name
+// Functionality: remove the object from the array that has the provided `name`.  Incorporate the findIndex method and the splice method.
+
+const deleteSubmissionByName = (array, name) => {
+    const index = array.findIndex((submission) => submission.name === name);
+    if (index !== -1) array.splice(index, 1);
+};
+deleteSubmissionByName(submissions, "Troy");
+console.log("Submissions After Deleting by Name", submissions);
+
+// specFive
+// Declare a function named editSubmission
+// Parameter(s): array, index, score
+// Functionality: update an object’s `score` in the `array` at the specified `index`. Use conditional statements to set the value for the `passed` property to `true` if the score is greater than or equal to 60 and `false` otherwise. No array method necessary.
+
+const editSubmission = (array, index, score) => {
+    array[index].score = score;
+    array[index].passed = score >= 60 ? true : false;
+};
+editSubmission(submissions, 0, 89);
+console.log("Editeded Submissions", submissions);
+// Changing Jane's score back to 95
+editSubmission(submissions, 0, 95);
+
+// specSix
+// Declare a function named findSubmissionByName
+// Parameter(s): array, name
+// Functionality: return the obejct in the `array` that has the provided `name`. Use the find method.
+
+const findSubmissionByName = (array, name) => array.find((sub) => sub.name === name);
+const findJane = findSubmissionByName(submissions, "Jane");
+console.log("Find Submission by Name", findJane);
+
+// findSubmissionByName(submissions, "Jane");
+// console.log("Find Submission by Name", submissions);  
+// Had troubles logging this one...
+
+// specSeven
+// Declare a function named findLowestScore
+// Parameter(s): array
+// Functionality: return the object in the `array` that has the lowest score.
+
+const findLowestScore = (array) => {
+    let lowestScore = 100;
+    array.forEach((sub) => {
+        const lowerScore = sub.score <= lowestScore ? true : false;
+        // console.log(lowerScore) testing if the forEach iterator works
+        if (lowerScore === true) {
+            lowestScore = sub.score
+        }
+    });
+    return lowestScore;
```

Comment:
Close, but you are supposed to return the entire object. Partly my bad, I remember helping you set this up. Here's a similar solution that returns the entire object: 

```
let lowestScore = array[0]; // Set the checker as the first object in the array
array.forEach((sub) => {  // loop over the elements of the array
    if (sub.score < lowestScore) {  // if the current element has a lower score than the lowestScore so far
        lowestScore = sub  // replace the lowestScore with the current element
        }
    }
return lowesScore;
```